modules/pipeline/dbclient/op_pipeline.go

```diff
@@ -352,45 +382,74 @@ func (client *Client) PageListPipelines(req apistructs.PipelinePageListRequest,
 		baseSQL.Alias(fmt.Sprintf("`%s` USE INDEX (%s)", (&spec.PipelineBase{}).TableName(), strings.Join(forceIndexes, ",")))
 	}
 
+	if req.PipelineDefinitionRequest != nil {
+		var definitionReq = req.PipelineDefinitionRequest
+		needQueryDefinition = true
+
+		baseSQL.Where(tableFieldName((&spec.PipelineBase{}).TableName(), "pipeline_definition_id") + " is not null ")
+		baseSQL.Where(tableFieldName((&spec.PipelineBase{}).TableName(), "pipeline_definition_id") + " != '' ")
+		if !definitionReq.IsEmptyValue() {
+			baseSQL.Join("INNER", definitiondb.PipelineDefinition{}.TableName(), fmt.Sprintf("%v.id = %v.pipeline_definition_id", definitiondb.PipelineDefinition{}.TableName(), (&spec.PipelineBase{}).TableName()))
+			baseSQL.Join("INNER", sourcedb.PipelineSource{}.TableName(), fmt.Sprintf("%v.id = %v.pipeline_source_id", sourcedb.PipelineSource{}.TableName(), definitiondb.PipelineDefinition{}.TableName()))
+			if len(definitionReq.Name) > 0 {
+				baseSQL.Where(fmt.Sprintf("%v.name like ?", definitiondb.PipelineDefinition{}.TableName()), definitionReq.Name+"%")
+			}
+			if len(definitionReq.SourceRemotes) > 0 {
+				baseSQL.In(tableFieldName(sourcedb.PipelineSource{}.TableName(), "remote"), definitionReq.SourceRemotes)
+			}
+			if len(definitionReq.Creators) > 0 {
+				baseSQL.In(tableFieldName(definitiondb.PipelineDefinition{}.TableName(), "creator"), definitionReq.Creators)
+			}
+		}
+	}
+
 	if !req.AllSources && len(req.Sources) > 0 {
-		baseSQL.In("pipeline_source", req.Sources)
+		baseSQL.In(tableFieldName((&spec.PipelineBase{}).TableName(), "pipeline_source"), req.Sources)
 	}
 	if len(req.YmlNames) > 0 {
-		baseSQL.In("pipeline_yml_name", req.YmlNames)
+		baseSQL.In(tableFieldName((&spec.PipelineBase{}).TableName(), "pipeline_yml_name"), req.YmlNames)
 	}
 	if len(req.Statuses) > 0 {
-		baseSQL.In("status", req.Statuses)
+		baseSQL.In(tableFieldName((&spec.PipelineBase{}).TableName(), "status"), req.Statuses)
 	}
 	if len(req.NotStatuses) > 0 {
-		baseSQL.NotIn("status", req.NotStatuses)
+		baseSQL.NotIn(tableFieldName((&spec.PipelineBase{}).TableName(), "status"), req.NotStatuses)
 	}
 	if len(req.TriggerModes) > 0 {
-		baseSQL.In("trigger_mode", req.TriggerModes)
+		baseSQL.In(tableFieldName((&spec.PipelineBase{}).TableName(), "trigger_mode"), req.TriggerModes)
 	}
 	if len(req.ClusterNames) > 0 {
-		baseSQL.In("cluster_name", req.ClusterNames)
+		baseSQL.In(tableFieldName((&spec.PipelineBase{}).TableName(), "cluster_name"), req.ClusterNames)
 	}
-	baseSQL.Where("is_snippet = ?", req.IncludeSnippet)
+	baseSQL.Where(tableFieldName((&spec.PipelineBase{}).TableName(), "is_snippet")+" = ?", req.IncludeSnippet)
```

Comment:
why there is a "+" somewhere?

Reply:
Join query is added, and all corresponding query fields must be spliced with table names